src/systems/systems/src/enhanced_gpwc/mod.rs

```diff
@@ -0,0 +1,193 @@
+            self.longitude =
+                Arinc429Word::new(Angle::new::<degree>(0.0), SignStatus::FailureWarning);
+            self.altitude = Arinc429Word::new(Length::new::<foot>(0.0), SignStatus::FailureWarning);
+            self.heading = Arinc429Word::new(Angle::new::<degree>(0.0), SignStatus::FailureWarning);
+            self.vertical_speed = Arinc429Word::new(
+                Velocity::new::<foot_per_minute>(0.0),
+                SignStatus::FailureWarning,
+            );
+        }
+    }
+
+    pub fn update(
+        &mut self,
+        adirs_output: &impl AdirsMeasurementOutputs,
+        lgcius: &LandingGearControlInterfaceUnitSet,
+    ) {
+        self.update_position_data(adirs_output);
+        self.gear_is_down = lgcius.lgciu1().main_down_and_locked();
+
+        self.navigation_displays.iter_mut().for_each(|display| {
+            display.update(
+                &self.navigation_display_range_lookup,
+                adirs_output.is_fully_aligned(1),
+            )
+        });
+    }
+}
+
+impl SimulationElement for EnhancedGPWC {
+    fn receive_power(&mut self, buses: &impl ElectricalBuses) {
+        self.is_powered = buses.is_powered(self.powered_by)
+    }
+
+    fn read(&mut self, reader: &mut SimulatorReader) {
+        let destination_long: f64 = reader.read(&self.fm1_destination_longitude_id);
+        let destination_lat: f64 = reader.read(&self.fm1_destination_latitude_id);
+        let destination_long_ssm: u32 = reader.read(&self.fm1_destination_longitude_ssm_id);
+        let destination_lat_ssm: u32 = reader.read(&self.fm1_destination_latitude_ssm_id);
+
+        self.destination_longitude = Arinc429Word::new(
+            Angle::new::<degree>(destination_long),
+            SignStatus::from(destination_long_ssm),
+        );
+        self.destination_latitude = Arinc429Word::new(
+            Angle::new::<degree>(destination_lat),
+            SignStatus::from(destination_lat_ssm),
+        );
+    }
+
+    fn write(&self, writer: &mut SimulatorWriter) {
+        writer.write_arinc429(
+            &self.egpwc_destination_longitude_id,
+            self.destination_longitude.value().get::<degree>(),
+            self.destination_longitude.ssm(),
+        );
+        writer.write_arinc429(
+            &self.egpwc_destination_latitude_id,
+            self.destination_latitude.value().get::<degree>(),
+            self.destination_latitude.ssm(),
+        );
+        writer.write_arinc429(
+            &self.egpwc_present_latitude_id,
+            self.latitude.value().get::<degree>(),
+            self.latitude.ssm(),
+        );
+        writer.write_arinc429(
+            &self.egpwc_present_longitude_id,
+            self.longitude.value().get::<degree>(),
+            self.longitude.ssm(),
+        );
+        writer.write(&self.egpwc_gear_is_down_id, self.gear_is_down);
+    }
+
+    fn accept<T: SimulationElementVisitor>(&mut self, visitor: &mut T) {
+        self.navigation_displays
```

Comment:
**suggestion**: `accept_iterable!(self.navigation_displays, visitor);`

Reply:
Will adapt it. Didn't know that the feature exists